### PR TITLE
Trialling `muffet` for link checking

### PR DESCRIPTION
**Pull request by lachlanhardy**
This resolves DOC-313:
https://linear.app/buildkite/issue/DOC-313/better-link-checker

### CODE REVIEW

**Comment by karensawrey** [comment]
Is muffet failing a sign it works or vice versa? :D If you need a +1 to merge it when it functions as it says on the label, @ me.

**Comment by lachlanhardy** [comment]
@karensawrey Thanks! Depends on what the failure is ;) But yeah, it's supposed to pass. Right now I'm having issues connecting Docker things. Very much not my area.

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -14,6 +14,20 @@ steps:
...
 
+  - label: ":spider: muffet"
+    depends_on: "rspec"
+    command: http://app:3000/docs --include=/docs/ --exclude=https://github.com/buildkite/docs/ --max-connections=10 --color=always --verbose
+    env:
+      RAILS_ENV: test
+    plugins:
+      - docker-compose#v3.0.0:
+          run: muffet
+          shell: false
```
Comment (matthewborden) [inline .buildkite/pipeline.yml:25]:
```suggestion
          shell: false
          upload-container-logs: always
```

**Comment by plaindocs** [comment]
@lachlanhardy lemme know if you want me to revert any of that (or feel free to yourself)

**Pull request closed without merging**